### PR TITLE
[release-4.4] Bug 1843230: Read BMH status from annotation

### PR DESCRIPTION
**Pull request by zaneb**
Backport of #68 to release-4.4, so so we can populate masters' IP information to make the node/machine link.

### CODE REVIEW

File: pkg/controller/baremetalhost/baremetalhost_controller.go [hunk 1]
```diff
@@ -168,6 +170,23 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
...
+		reqLogger.Info("Fetching Status from Annotation")
+		objStatus, err := r.getHostStatusFromAnnotation(host)
+		if err == nil && objStatus != nil {
+			host.Status = *objStatus
+			errStatus := r.client.Status().Update(context.TODO(), host)
+			if errStatus != nil {
+				return reconcile.Result{}, errors.Wrap(err, "Could not restore status from annotation")
+			}
+			return reconcile.Result{Requeue: true}, nil
+		} else {
```
Comment (stbenjam) [inline pkg/controller/baremetalhost/baremetalhost_controller.go:185]: This else should be dropped:

```
./pkg/controller/baremetalhost/baremetalhost_controller.go:185:10: if block ends with a return statement, so drop this else and outdent its block 
```

**Comment by dhellmann** [comment]
This is going to need to include https://github.com/metal3-io/baremetal-operator/pull/546 as well

**Comment by zaneb** [comment]
The 4.5 bugzilla is still ON_QA, so it won't let us merge a backport to 4.4 until that is verified.

**Pull request merged**